redash/query_runner/big_query.py

```diff
@@ -332,6 +333,18 @@ def run_query(self, query, user):
                 error = json_loads(e.content)["error"]["message"]
             else:
                 error = e.content
+        except (KeyboardInterrupt, InterruptException, JobTimeoutException):
+            if self.current_job_id:
+                response_kind = None
+                while response_kind != "bigquery#jobCancelResponse":
+                    response_kind = jobs.cancel(
```

Comment:
We need to repeatedly execute the cancel request to get its status? No way to just get status after the first request?

Reply:
This bothered me as well. Turns out the client is executing previous requests when calling `execute()` in the cancellation block (the result kind is a `getQueryResultResponse` for a call to `/cancel` 🤨), so I'm guessing it's either some sort of optimization or a bug. Tried drilling into their source code to see if this is intentional, and if there's a flush method of some sort. Eventually gave up and realized it'll be easier to instantiate a new client. See 893c427.